htdocs/core/lib/invoice.lib.php

```diff
@@ -1333,3 +1333,91 @@ function getPurchaseInvoiceUnpaidOpenTable($maxCount = 500, $socid = 0)
 
 	return $result;
 }
+
+/**
+ * Calculate Creditor Reference RF / FI Bank payment reference number
+ *
+ * @param 	string	$invoice_number	    Invoice number to generate payment reference
+ * @param 	int     $statut             Invoice status, if draft (0), no reference generating
+ * @param   string  $use_rf             false/(0) generate FI Bank payment reference
+ *                                      true/(1) Generate European Reference number based on the global
+ *                                      Structured Creditor Reference standard (SEPA RF creditor reference)
+ * @return 	string                      String Payment reference number or RF creditor reference
+ */
+function calculate_payment_reference($invoice_number, $statut, $use_rf)
```

Comment:
Can you tell me for which country to you need this function ?
In which part of code (internal / external ?) do you plan to call this method ?

Reply:
Mainly in Finland. Also it is for european global RF Creditor Reference number for payments.
https://www.finanssiala.fi/wp-content/uploads/2021/03/Introduction_of_the_global_Structured_Creditor_Reference.pdf
Also for generate barcode for paying with Finnish refererence and the international (RF) reference standard format.
https://www.finanssiala.fi/wp-content/uploads/2021/03/Bank_bar_code_guide.pdf
Second phase there is QR code for global european payments.
https://www.finanssiala.fi/wp-content/uploads/2021/03/QR_code_in_credit_transfer_forms.pdf